lua/wikis/mobilelegends/Infobox/Character/Custom.lua

```diff
@@ -0,0 +1,214 @@
+---
+-- @Liquipedia
+-- wiki=mobilelegends
+-- page=Module:Infobox/Character/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local CharacterWinLoss = require('Module:CharacterWinLoss')
+local Class = require('Module:Class')
+local CostDisplay = require('Module:CostDisplay')
+local Flags = require('Module:Flags')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Math = require('Module:MathUtil')
+local Namespace = require('Module:Namespace')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+
+local Character = Lua.import('Module:Infobox/Character')
+local Injector = Lua.import('Module:Widget/Injector')
+
+local Widgets = Lua.import('Module:Widget/All')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Cell = Widgets.Cell
+local IconImageWidget = Lua.import('Module:Widget/Image/Icon/Image')
+local Title = Widgets.Title
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+---@class MobileLegendsHeroInfobox: CharacterInfobox
+local CustomCharacter = Class.new(Character)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return Html
+function CustomCharacter.run(frame)
+	local character = CustomCharacter(frame)
+	character:setWidgetInjector(CustomInjector(character))
+	character.args.informationType = 'Hero'
+	return character:createInfobox()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	if id == 'country' then
+		return {
+			Cell{name = 'Region', content = {self.caller.args.region}},
+			Cell{name = 'City', content = {self.caller.args.city}},
+		}
+	elseif id == 'role' then
+		return {
+			Cell{
+				name = 'Role',
+				content = WidgetUtil.collect(
+					self:_toCellContent('primaryrole', 'ClassIcon'),
+					self:_toCellContent('secondaryrole', 'ClassIcon')
+				)
+			},
+			Cell{
+				name = 'Lane',
+				content = {self:_toCellContent('lane', 'ClassIcon')}
+			},
+		}
+	elseif id == 'custom' then
+		return WidgetUtil.collect(
+			self.caller:_getPriceCell(),
+			self.caller:_getCustomCells()
+		)
+	end
+
+	return widgets
+end
+
+---@param key string
+---@param dataModule string
+---@return Widget?
+function CustomInjector:_toCellContent(key, dataModule)
```

Comment:
this only ever gets called with the same dataModule
imho imnport the data module at top and kick the 2nd passed var entirely